src/test/java/com/softserve/academy/spaced/repetition/controller/CourseControllerTest.java

```diff
@@ -63,196 +73,96 @@ public void getCourseById() throws Exception {
-        category.setId(3L);
-        course.setCategory(category);
-
-        Course course2 = new Course();
-        course2.setId(2L);
-        course2.setName("C++ interview course");
-        course2.setDescription("3 parts of java questions & answers");
-        course2.setRating(0);
-        course2.setImage(new Image(15L));
-        course2.setPublished(true);
-        course2.setOwner(user);
-        Category category2 = new Category();
-        category2.setId(2L);
-        course2.setCategory(category2);
+        Course course = createCourse(3L,"C# interview course", "questions & answers",
+                0,16L,true,1L,"admin@gmail.com", 1,3L);
 
-        Course course3 = new Course();
-        course3.setId(1L);
-        course3.setName("Java interview course");
-        course3.setDescription("4 parts of java questions & answers");
-        course3.setRating(0);
-        course3.setImage(new Image(14L));
-        course3.setPublished(true);
-        course3.setOwner(user);
-        Category category3 = new Category();
-        category3.setId(1L);
-        course3.setCategory(category3);
+        Course course2 = createCourse(2L,"C++ interview course", "3 parts of java questions & answers",
+                0,15L,true,1L,"admin@gmail.com",1,2L);
 
-        Course course4 = new Course();
-        course4.setId(5L);
-        course4.setName("JavaScript interview course");
-        course4.setDescription("questions & answers");
-        course4.setRating(0);
-        course4.setImage(new Image(18L));
-        course4.setPublished(true);
-        course4.setOwner(user);
-        Category category4 = new Category();
-        category4.setId(10L);
-        course4.setCategory(category4);
+        Course course3 = createCourse(1L,"Java interview course", "4 parts of java questions & answers",
+                0,14L,true,1L,"admin@gmail.com",1,1L);
 
-        Course course5 = new Course();
-        course5.setId(4L);
-        course5.setName("PHP interview course");
-        course5.setDescription("2 parts of java questions & answers");
-        course5.setRating(0);
-        course5.setImage(new Image(17L));
-        course5.setPublished(true);
-        course5.setOwner(user);
-        Category category5 = new Category();
-        category5.setId(4L);
-        course5.setCategory(category5);
+        Course course4 = createCourse(5L,"JavaScript interview course", "questions & answers",
+                0,18L,true,1L,"admin@gmail.com",1,10L);
 
+        Course course5 = createCourse(4L,"PHP interview course", "2 parts of java questions & answers",
+                0,17L,true,1L,"admin@gmail.com",1,4L);
 
         courseList.add(course);
         courseList.add(course2);
         courseList.add(course3);
         courseList.add(course4);
         courseList.add(course5);
 
-        Page<Course> coursePage = new PageImpl<>(courseList);
-        return coursePage;
+        Page<Course> coursesPage;
+        if(ASCENDING) {
```

Comment:
This was commented on the demo but you are still doing the same mistake.
I would start by introducing variable 
`Sort.Direction direction = ASCENDING ? Sort.Direction.ASC : Sort.Direction.DESC;`

Please find the code that was commented on demo and improve.

By the way this if doesn't make sense as ASCENDING is constant and will always have the same value.